libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Action/CSFSalesforceAction.m

```diff
@@ -324,18 +296,37 @@ + (BOOL)isNetworkError:(nullable NSError *)error {
 }
 
 #pragma mark SFAuthenticationManagerDelegate
-
-- (void)userAccountManagerDidChangeCurrentUser:(NSNotification*)notification {
+- (void)userAccountManagerDidChangeUserDataNotification:(NSNotification*)notification {
     SFUserAccountManager *accountManager = (SFUserAccountManager*)notification.object;
+
     if ([accountManager isKindOfClass:[SFUserAccountManager class]]) {
-        if (![accountManager.currentUserIdentity isEqual:self.enqueuedNetwork.account.accountIdentity]) {
-            self.enqueuedNetwork.networkSuspended = YES;
-        } else {
-            [self.enqueuedNetwork resetSession];
-            self.enqueuedNetwork.networkSuspended = NO;
-        }
-        if (accountManager.currentCommunityId != self.enqueuedNetwork.defaultConnectCommunityId) {
-            self.enqueuedNetwork.defaultConnectCommunityId = accountManager.currentCommunityId;
+
+        NSString *userId = notification.userInfo[SFUserAccountManagerUserChangeUserIdKey];
+        NSString *orgId = notification.userInfo[SFUserAccountManagerUserChangeOrgIdKey];
+        SFUserAccountIdentity *identity = [[SFUserAccountIdentity alloc] initWithUserId:userId orgId:orgId];
+        SFUserAccount *userAccount = [[SFUserAccountManager sharedInstance] userAccountForUserIdentity:identity];
+        SFUserAccountChange change = (SFUserAccountChange)[notification.userInfo[SFUserAccountManagerUserChangeKey] integerValue];
+
+        if (change & SFUserAccountChangeCurrentUser) {
```

Comment:
Is there any reason why the salesforce actions need to know that the "Current User" changes?  The network instances shouldn't care who the current user is; rather, if the application using the network decides to make network requests after switching a user, it's their prerogative to utilize the network via that new user.